uwblueprint/sistering
# Create frontend utilities using Moment.js

## Ticket link

Closes #145 



## Implementation description
* created 3 moment helper functions in the file DateTimeUtils.ts
* formatTimeHourMinutes and getElapsedHours have been implemented in VolunteerAvailabilityTableRow.tsx to display the time as (5:00pm) and calculate the difference in hours between two given times
* <img width="345" alt="Screen Shot 2022-03-04 at 7 06 40 PM" src="https://user-images.githubusercontent.com/82241706/156865441-bc3653c6-aec9-409c-9a7b-24420d8b819b.png">
* formatTimeHourMinutes can also be used to display the start and end times for events in PostingDetails.tsx and PostingCard.tsx

* formatDateStringYear have been implemented in PostingDetails.tsx and PostingCard.tsx to display the start and end dates in the format (Monday, Feb 20, 2021)
<img width="286" alt="Screen Shot 2022-03-04 at 7 09 13 PM" src="https://user-images.githubusercontent.com/82241706/156865531-7c7205ed-027c-41fa-bdf5-7572e89632a4.png">
<img width="333" alt="Screen Shot 2022-03-04 at 7 09 33 PM" src="https://user-images.githubusercontent.com/82241706/156865532-72204f4c-ed13-4aeb-a386-00985fbb899c.png">

* Looking at the Figma, it can also be used in "Requests Pending Confirmation" 
<img width="419" alt="Screen Shot 2022-03-04 at 7 17 38 PM" src="https://user-images.githubusercontent.com/82241706/156865739-64eeef09-1a02-45e9-920a-500e72540a7d.png">



## Steps to test
Navigate to http://localhost:5000/graphql and run the mutations (see: https://github.com/uwblueprint/sistering/pull/140)

Navigate to http://localhost:3000/volunteer/postings to check if formatDateStringYear formats the dates correctly
Navigate to http://localhost:3000/volunteer/posting/1 to check if formatDateStringYear formats the dates correctly
To test formatTimeHourMinutes and getElapsedHours, the [VolunteerAvailabilityTableRow](https://github.com/uwblueprint/sistering/pull/134) component must be added to an existing page 


## What should reviewers focus on?
1. If the current implementations are appropriate
2. If there are more places to use them 
3. If other moment.js utilities need to be made


## Checklist
- [ x] My PR name is descriptive and in imperative tense
- [ x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ x] I have run the appropriate linter(s)
- [ x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
